src/content/popup/swipe.ts

```diff
@@ -0,0 +1,59 @@
+type HorizontalSwipeDirection = 'right' | 'left';
+
+export function onHorizontalSwipe(
+  element: HTMLElement,
+  handler: (swipeDirection: HorizontalSwipeDirection) => void
+) {
+  // Min x distance traveled to be considered swipe
+  const xMinThreshold = 50;
+  // Max y distance that can be traveled before
+  // it's no longer considered a horizontal swipe
+  const yMaxThreshold = 100;
+  // Max time allowed to travel that distance
+  const allowedTime = 200;
+
+  let startTime = 0;
+  let startX: number;
+  let startY: number;
+
+  element.addEventListener(
+    'touchstart',
+    function (e) {
+      startX = e.changedTouches[0].pageX;
+      startY = e.changedTouches[0].pageY;
+      startTime = Date.now();
```

Comment:
Can we use `performance.now()` instead of `Date.now()` here and below? It seems like it should be better suited to this ([ref](https://developer.mozilla.org/docs/Web/API/Performance/now#performance.now_vs._date.now)).